resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -252,11 +252,11 @@ else if (valueOf == null)
    }
    
 	private Class<? extends Collection> convertParameterTypeToCollectionType() {
-		if (List.class.isAssignableFrom(type)) {

```

Comment:
This worries me. I think item 5 is a **bit** ambigious, since ArrayList **is** a List, though I think your interpretation is more plausible. But, since StringParameterInjector.extractValues() returns a HashSet, 

> @GET
>public Response singleValuedQueryParam(@QueryParam("date") HashSet dates) {
>...
>}

should work, right? So we're excluding behavior that used to work. That's not good.